wireapp/wire-ios-mono
# fix: backup restore events not tracked - WPB-11308

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-11308" title="WPB-11308" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />WPB-11308</a>  [iOS] Backup Restore Succeeded event isn't being logged in Countly
  </td></table>
  <br />
 













### Issue

Backup restore (success and failure) events are not tracked. The events were being sent through the (optional) active user session, but at that point in time there doesn't exist an active user session. The solution is to pass the result of the backup import along to the point where an active user session exists and send the events via the analytics service:

- `BackupRestoreController` passes the result to its delegate (`AuthenticationCoordinator`)
- `AuthenticationCoordinator` enqueues the action `completeBackupStep(didSucceed: Bool?)`
- `AuthenticationCoordinator` executes the action, telling the `UnauthenticatedSession` to report the result. 
- `UnauthenticatedSession` will `upgradeToAuthenticatedSession`, then inform its delegate (`SessionManager`)
- `SessionManager` will create and activate `ZMUserSession`, then send an event to the user session via `userSession.trackAnalyticsEvent(.backupRestored)`.
- If `ZMUserSession` has been configured to send analytics events, it will forward the event to `analyticsEventTracker`.
- Otherwise it add events to `pendingAnalyticsEvents`, and when the `analyticsEventTracker` is set it will forward the pending events.

### Testing

- Log in, restore a backup successfully, enable analytics, check the event is tracked
- Log in, fail to restore a backup (insert wrong password), continue without backup, enable analytics, check the event is tracked

---

### Checklist


---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
